scipy/io/_fast_matrix_market/fast_matrix_market/include/fast_matrix_market/app/array.hpp

```diff
@@ -8,7 +8,7 @@
 
 namespace fast_matrix_market {
 
-#if __cplusplus >= 202002L
+#if __cplusplus >= 202002L || (defined(_MSVC_LANG) && _MSVC_LANG >= 202002L)
```

Comment:
The only [check](https://github.com/search?q=repo%3Ascipy%2Fscipy%20%2F_MSVC_LANG%2F&type=code) we currently have for `_MSVC_LANG` does a `+0L` to avoid the `defined` check - while this one is cleaner, I'm not sure we gain much for it. And also, I'd like to document the reason again:
```suggestion
// MSVC does not define __cplusplus correctly by default; fall back to _MSVC_LANG if it exists
#if __cplusplus >= 202002L || _MSVC_LANG+0L >= 202002L
```

Reply:
Normally I check for definition to avoid the -Wundef warning, but I assume that's not a major concern.

Reply:
We try hard to keep the build clean of warnings. The original here looks both cleaner and more readable; why was there a need to change it?

Reply:
The latest commit reverts to the above, and changes the other instance of `_MSVC_LANG+0L` for the desired consistency.